compiler/noirc_frontend/src/lexer/token.rs

```diff
@@ -948,6 +952,14 @@ impl SecondaryAttribute {
             SecondaryAttribute::Abi(_) => Some("abi".to_string()),
             SecondaryAttribute::Varargs => Some("varargs".to_string()),
             SecondaryAttribute::UseCallersScope => Some("use_callers_scope".to_string()),
+            SecondaryAttribute::Allow(_) => Some("allow".to_string()),
+        }
+    }
+
+    pub(crate) fn is_allow_unused_variables(&self) -> bool {
+        match self {
+            SecondaryAttribute::Allow(string) => string == "unused_variables",
```

Comment:
Do we also need to prevent someone making a function that they would expect to be usable as a `SecondaryAttribute` named `unused_variables`? (I.e. with a clear error)

Reply:
Maybe, but so far this isn't done in any other of the other attributes. Or... I guess you could define a function named like that, it's just that you won't be able to use it to run macro code.

Reply:
Done! Now attributes are only parsed in a let declaration in statements, never in a trait.